sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -303,12 +303,27 @@ private case class PostgresDialect()
 
   class PostgresSQLBuilder extends JDBCSQLBuilder {
     override def visitExtract(field: String, source: String): String = {
-      field match {
-        case "DAY_OF_YEAR" => s"EXTRACT(DOY FROM $source)"
-        case "YEAR_OF_WEEK" => s"EXTRACT(YEAR FROM $source)"
-        case "DAY_OF_WEEK" => s"EXTRACT(DOW FROM $source)"
-        case _ => super.visitExtract(field, source)
+      // SECOND, MINUTE, HOUR, QUARTER, YEAR, DAY are identical on postgres and spark
+      // MONTH        is different, postgres returns 0-11, spark returns 1-12.
```

Comment:
I checked the doc you provided, but it says 1-based for `datetimes`, 0-based for `intervals` on the pg side.

Reply:
Yeah, I believe they shall be identical as they all came from the work within the Postgres feature parity umbrella ticket.

I appreciate the confirmation.